Assets/MixedRealityToolkit/Inspectors/MixedRealityShaderGUI.cs

```diff
@@ -0,0 +1,274 @@
+            public static string colorWriteMaskName = "_ColorWriteMask";
+
+            public static string alphaTestOnName = "_ALPHATEST_ON";
+            public static string alphaBlendOnName = "_ALPHABLEND_ON";
+
+            public static readonly string[] renderingModeNames = Enum.GetNames(typeof(RenderingMode));
+            public static readonly string[] customRenderingModeNames = Enum.GetNames(typeof(CustomRenderingMode));
+            public static readonly string[] depthWriteNames = Enum.GetNames(typeof(DepthWrite));
+            public static GUIContent sourceBlend = new GUIContent("Source Blend", "Blend Mode of Newly Calculated Color");
+            public static GUIContent destinationBlend = new GUIContent("Destination Blend", "Blend Mode of Existing Color");
+            public static GUIContent blendOperation = new GUIContent("Blend Operation", "Operation for Blending New Color With Existing Color");
+            public static GUIContent depthTest = new GUIContent("Depth Test", "How Should Depth Testing Be Performed.");
+            public static GUIContent depthWrite = new GUIContent("Depth Write", "Controls Whether Pixels From This Material Are Written to the Depth Buffer");
+            public static GUIContent depthOffsetFactor = new GUIContent("Depth Offset Factor", "Scales the Maximum Z Slope, with Respect to X or Y of the Polygon");
+            public static GUIContent depthOffsetUnits = new GUIContent("Depth Offset Units", "Scales the Minimum Resolvable Depth Buffer Value");
+            public static GUIContent colorWriteMask = new GUIContent("Color Write Mask", "Color Channel Writing Mask");
+            public static GUIContent cullMode = new GUIContent("Cull Mode", "Triangle Culling Mode");
+            public static GUIContent renderQueueOverride = new GUIContent("Render Queue Override", "Manually Override the Render Queue");
+        }
+
+        protected static void SetupMaterialWithRenderingMode(Material material, RenderingMode mode, CustomRenderingMode customMode, int renderQueueOverride)
+        {
+            switch (mode)
+            {
+                case RenderingMode.Opaque:
+                    {
+                        material.SetOverrideTag(BaseStyles.renderTypeName, BaseStyles.renderingModeNames[(int)RenderingMode.Opaque]);
+                        material.SetInt(BaseStyles.customRenderingModeName, (int)CustomRenderingMode.Opaque);
+                        material.SetInt(BaseStyles.sourceBlendName, (int)BlendMode.One);
+                        material.SetInt(BaseStyles.destinationBlendName, (int)BlendMode.Zero);
+                        material.SetInt(BaseStyles.blendOperationName, (int)BlendOp.Add);
+                        material.SetInt(BaseStyles.depthTestName, (int)CompareFunction.LessEqual);
+                        material.SetInt(BaseStyles.depthWriteName, (int)DepthWrite.On);
+                        material.SetFloat(BaseStyles.depthOffsetFactorName, 0.0f);
```

Comment:
Aside from the Custom rendering mode, several of these material BaseStyles end up having the same line of code/configuration in all of these options (i.e. 0.0f for this and depthOffsetUnitsName, depthTestName for LessEqual, and so on.

I can see value in going either way (i.e. having it duplicated explicitly shows for each configuration what is expected, having the common stuff taken out helps to highlight the deltas between each configuration)

I just wanted to ask about the intent and check that that was the goal, cause if so that's all good.

Reply:
I only ask because usually i see copy+paste = bad, but in some cases it's fine if the goal is being explicit about configuration.

Reply:
@wiwei this is actually the original implementation that I moved to the base class. @Cameron-Micka , assuming you made this, is there any reason not to simplify?

Reply:
The original inspiration came from [Unity's implementation](https://github.com/Unity-Technologies/UnityCsReference/blob/053d3ce1c53ee3dd4ff1aa08522c1284453f0796/Editor/Mono/Inspector/StandardShaderGUI.cs#L335) but since I've added features it has ballooned a little bit. I somewhat like the explicitness so I wouldn't mind keeping it as is, but feel free to factor out the common properties as long as when the user selects "Custom" it retains their previous settings.

Reply:
TL;DR It's your call. ;)

Reply:
I agree with removing out the duplicated properties. Slims down the total file size and again keeps editing if need be in the future to one location instead of 5